setup.py

```diff
@@ -238,7 +238,7 @@ def _get_arrow_lib_as_linker_input(self):
         ],
         "pandas": pandas_requirements,
         "development": [
-            'pytest<6.1.0',
+            'pytest',
```

Comment:
can you add an upper bound?

Reply:
This is only used for testing, I'd like to be on the bleeding edge for testing.
Unless of course things are broken, which they were in this case.

Reply:
Can't we keep the upper bound and have dependabot submit PRs when there is a new version? Having this upper bound would prevent our test pipeline from breaking if pytest makes a breaking change

Reply:
Okay let's do it, if you guys think that it is useful 👍